sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -70,14 +77,15 @@ class OkHttpAsyncHttpClient implements HttpClient {
     private static final byte[] EMPTY_BODY = new byte[0];
     private static final RequestBody EMPTY_REQUEST_BODY = RequestBody.create(EMPTY_BODY);
 
-    private static final String AZURE_EAGERLY_READ_RESPONSE = "azure-eagerly-read-response";
-    private static final String AZURE_IGNORE_RESPONSE_BODY = "azure-ignore-response-body";
-    private static final String AZURE_EAGERLY_CONVERT_HEADERS = "azure-eagerly-convert-headers";
-
     final OkHttpClient httpClient;
 
-    OkHttpAsyncHttpClient(OkHttpClient httpClient) {
-        this.httpClient = httpClient;
+    private final Duration responseTimeout;
+
+    OkHttpAsyncHttpClient(OkHttpClient httpClient, Duration responseTimeout) {
```

Comment:
how bad would it be if we built a new per-call client (if per call timeout is set) according to okhttp recepies https://github.com/square/okhttp/blob/master/samples/guide/src/main/java/okhttp3/recipes/PerCallSettings.java ?

maybe if there is enough caching and pooling the perf impact would be comparable/lower than a time-base solution?

Reply:
oh snap